homeassistant/components/asuswrt/bridge.py

```diff
@@ -0,0 +1,289 @@
+        """Get list of connected devices."""
+        try:
+            api_devices = await self._api.async_get_connected_devices()
+        except OSError as exc:
+            raise UpdateFailed(exc) from exc
+        return {
+            format_mac(mac): WrtDevice(dev.ip, dev.name, None)
+            for mac, dev in api_devices.items()
+        }
+
+    async def _get_nvram_info(self, info_type: str) -> dict[str, Any]:
+        """Get AsusWrt router info from nvram."""
+        info = {}
+        try:
+            info = await self._api.async_get_nvram(info_type)
+        except OSError as exc:
+            _LOGGER.warning(
+                "Error calling method async_get_nvram(%s): %s", info_type, exc
+            )
+
+        return info
+
+    async def _get_label_mac(self) -> None:
+        """Get label mac information."""
+        label_mac = await self._get_nvram_info("LABEL_MAC")
+        if label_mac and "label_mac" in label_mac:
+            self._label_mac = format_mac(label_mac["label_mac"])
+
+    async def _get_firmware(self) -> None:
+        """Get firmware information."""
+        firmware = await self._get_nvram_info("FIRMWARE")
+        if firmware and "firmver" in firmware:
+            firmver: str = firmware["firmver"]
+            if "buildno" in firmware:
+                firmver += f" (build {firmware['buildno']})"
+            self._firmware = firmver
+
+    async def _get_model(self) -> None:
+        """Get model information."""
+        model = await self._get_nvram_info("MODEL")
+        if model and "model" in model:
+            self._model = model["model"]
+
+    async def async_get_available_sensors(self) -> dict[str, dict[str, Any]]:
+        """Return a dictionary of available sensors for this bridge."""
+        sensors_temperatures = await self._get_available_temperature_sensors()
+        sensors_types = {
+            SENSORS_TYPE_BYTES: {
+                KEY_SENSORS: SENSORS_BYTES,
+                KEY_METHOD: self._get_bytes,
+            },
+            SENSORS_TYPE_LOAD_AVG: {
+                KEY_SENSORS: SENSORS_LOAD_AVG,
+                KEY_METHOD: self._get_load_avg,
+            },
+            SENSORS_TYPE_RATES: {
+                KEY_SENSORS: SENSORS_RATES,
+                KEY_METHOD: self._get_rates,
+            },
+            SENSORS_TYPE_TEMPERATURES: {
+                KEY_SENSORS: sensors_temperatures,
+                KEY_METHOD: self._get_temperatures,
+            },
+        }
+        return sensors_types
+
+    async def _get_available_temperature_sensors(self) -> list[str]:
+        """Check which temperature information is available on the router."""
+        try:
+            availability = await self._api.async_find_temperature_commands()
+            available_sensors = [
+                SENSORS_TEMPERATURES[i] for i in range(3) if availability[i]
+            ]
+        except Exception as exc:  # pylint: disable=broad-except
```

Comment:
Is catching `Excetion` necessary here? Maybe `IndexError` and `OSError` will suffice?

Reply:
Not sure about this. Looking at the [code](https://github.com/kennedyshead/aioasuswrt/blob/0409d74161e14da6b9ebc05c8ee43b7e1fce48f4/aioasuswrt/asuswrt.py#L561) `IndexError` and `OsError` are already catch in the internal loop, so these exceptions should not be raised, but I don't know if other exceptions could be raised by the command.

Reply:
Hmm the result of `async_find_temperature_commands()` is always a list with three bool values and `IndexError`, `OSError` and `ValueError` are caught in the backend library. I don't think we need try..except here at all.

Reply:
Ok, I remove this, eventually we can add again specific exception if we got some issue

Reply:
Yes, if there are issue reports during the beta period, we will revert exception catching here.